SharedCode/src/commonMain/kotlin/http/APIErrors.kt

```diff
@@ -0,0 +1,6 @@
+package com.jetbrains.handson.mpp.mobile.http
+
+enum class APIErrors(val message: String) {
+    RequestError("Failed to fetch train times"),
+    SameStation("The departure station cannot be the same as the destination")
```

Comment:
This PR looks fine for what its aims are, but there are a couple more errors that I think we should handle explicitly (in a later PR?):

- User has not selected a station
- User has typed in an invalid selection (side note, currently if you type in "York" but don't select it in either android or iOS the API call fails as a Station has not been selected, only a String input)

Reply:
I agree we should add handling for more cases, but yeah figured this text should just be the basics.

For those specific examples though do you know where the error is actually happening - it might be occurring before the code even gets to the API level. Might be a case of handling those errors on the ui side.

Reply:
True, these are both UI-side errors